src/tracing/esp32_diagnostic_trace/DiagnosticTracing.cpp

```diff
@@ -0,0 +1,169 @@
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <algorithm>
+#include <esp_err.h>
+#include <esp_log.h>
+#include <tracing/esp32_diagnostic_trace/Counter.h>
+#include <tracing/esp32_diagnostic_trace/DiagnosticTracing.h>
+
+namespace chip {
+namespace Tracing {
+namespace Diagnostics {
+
+namespace Utils {
+// Implements a murmurhash with 0 seed.
+uint32_t MurmurHash(const void * key)
+{
+    const uint32_t kMultiplier = 0x5bd1e995;
+    const uint32_t kShift      = 24;
+    const unsigned char * data = (const unsigned char *) key;
+    uint32_t hash              = 0;
+
+    while (*data)
+    {
+        uint32_t value = *data++;
+        value *= kMultiplier;
+        value ^= value >> kShift;
+        value *= kMultiplier;
+        hash *= kMultiplier;
+        hash ^= value;
+    }
+
+    hash ^= hash >> 13;
+    hash *= kMultiplier;
+    hash ^= hash >> 15;
+
+    if (hash == 0)
+    {
+        ESP_LOGW("Tracing", "MurmurHash resulted in a hash value of 0");
+    }
+
+    return hash;
+}
+} // namespace Utils
+
+constexpr size_t kPermitListMaxSize = CONFIG_MAX_PERMIT_LIST_SIZE;
+using HashValue                     = uint32_t;
+using namespace Utils;
+
+HashValue gPermitList[kPermitListMaxSize] = { MurmurHash("PASESession"),
+                                              MurmurHash("CASESession"),
+                                              MurmurHash("NetworkCommissioning"),
+                                              MurmurHash("GeneralCommissioning"),
+                                              MurmurHash("OperationalCredentials"),
+                                              MurmurHash("CASEServer"),
+                                              MurmurHash("Fabric"),
+                                              MurmurHash("Resolver") }; // namespace
+
+bool IsPermitted(const char * str)
+{
+    HashValue hashValue = MurmurHash(str);
+    for (HashValue permitted : gPermitList)
+    {
+        if (permitted == 0)
+        {
+            break;
+        }
+        if (hashValue == permitted)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+void ESP32Diagnostics::LogMessageReceived(MessageReceivedInfo & info) {}
+
+void ESP32Diagnostics::LogMessageSend(MessageSendInfo & info) {}
+
+void ESP32Diagnostics::LogNodeLookup(NodeLookupInfo & info) {}
+
+void ESP32Diagnostics::LogNodeDiscovered(NodeDiscoveredInfo & info) {}
+
+void ESP32Diagnostics::LogNodeDiscoveryFailed(NodeDiscoveryFailedInfo & info) {}
+
+void ESP32Diagnostics::LogMetricEvent(const MetricEvent & event)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    switch (event.ValueType())
+    {
+    case ValueType::kInt32: {
+        ESP_LOGI("mtr", "The value of %s is %ld ", event.key(), event.ValueInt32());
```

Comment:
```suggestion
        ESP_LOGI("mtr", "The value of %s is %" PRId32, event.key(), event.ValueInt32());
```

I'd suggest using the width specific format specifiers, and it would be good to use `ChipLogProgress` instead of ESP_LOGx